app/registration.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Register components (via a list of glob patterns)
+ *
+ * Copyright © 2015 Magento. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\NonComposerComponentRegistration;
+
+use RuntimeException;
+
+/**
+ * Include files from a list of glob patterns
+ *
+ * @throws RuntimeException
+ * @return void
+ */
+function main()
+{
+    $globPatterns = require __DIR__ . '/etc/registration_globlist.php';
+    $baseDir = __DIR__ . '/';
+
+    foreach ($globPatterns as $globPattern) {
+        // Sorting is disabled intentionally for performance improvement
+        $files = glob($baseDir . $globPattern, GLOB_NOSORT);
+        if ($files === false) {
+            throw new RuntimeException("glob(): error with '$baseDir$globPattern'");
+        }
+        array_map(__NAMESPACE__ . '\file', $files);
```

Comment:
I was experimenting back and forth because of the problems in the build IIRC (integration testsuite). Usage should be local anyway, so usage might not be such a problem (and IIRC Phpstorm understands if a string represents a callback). In any way, it should be that variant taken that works best with the test-suite. I have no specific preference but *not* creating a function object on the fly (as suggested by @davidalger).